test/new-e2e/agent-platform/common/agent_behaviour.go

```diff
@@ -0,0 +1,203 @@
+	})
+
+	t.Run("datadog-agent checks running", func(tt *testing.T) {
+		var statusOutputJSON map[string]any
+		result := false
+		for try := 0; try < 5 && !result; try++ {
+			err := json.Unmarshal([]byte(client.Env.Agent.Status(e2eClient.WithArgs([]string{"-j"})).Content), &statusOutputJSON)
+			require.NoError(tt, err)
+			if runnerStats, ok := statusOutputJSON["runnerStats"]; ok {
+				runnerStatsMap := runnerStats.(map[string]any)
+				if checks, ok := runnerStatsMap["Checks"]; ok {
+					checksMap := checks.(map[string]any)
+					result = len(checksMap) > 0
+				}
+			}
+			time.Sleep(1 * time.Second)
+		}
+		require.True(tt, result, "status output should contain running checks")
+	})
+
+	t.Run("status command infos", func(tt *testing.T) {
+		statusOutput := client.Env.Agent.Status().Content
+		require.Contains(tt, statusOutput, "Forwarder")
+		require.Contains(tt, statusOutput, "Host Info")
+		require.Contains(tt, statusOutput, "DogStatsD")
+	})
+
+	t.Run("status command no errors", func(tt *testing.T) {
+		statusOutput := client.Env.Agent.Status().Content
+
+		// API Key is invalid we should not check for the following error
+		statusOutput = strings.ReplaceAll(statusOutput, "[ERROR] API Key is invalid", "API Key is invalid")
+		require.NotContains(tt, statusOutput, "ERROR")
+	})
+}
+
+// CheckAgentStops runs tests to check the agent can stop properly
+func CheckAgentStops(t *testing.T, client *ExtendedClient) {
+	t.Run("stops", func(tt *testing.T) {
+		_, err := client.SvcManager.Stop("datadog-agent")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Status("datadog-agent")
+		require.Error(tt, err, "datadog-agent service should be stopped")
+	})
+
+	t.Run("refuse connections", func(tt *testing.T) {
+		_, err := client.Env.Agent.StatusWithError()
+		require.Error(tt, err, "status should error")
+	})
+
+	t.Run("no running processes", func(tt *testing.T) {
+		agentProcesses := []string{"datadog-agent", "system-probe", "security-agent"}
+		for _, process := range agentProcesses {
+			_, err := client.Env.VM.ExecuteWithError(fmt.Sprintf("pgrep -f %s", process))
+			require.Error(tt, err, fmt.Sprintf("process %s should not be running", process))
+		}
+	})
+
+	t.Run("starts after stop", func(tt *testing.T) {
+		_, err := client.SvcManager.Start("datadog-agent")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Status("datadog-agent")
+		require.NoError(tt, err, "datadog-agent should be running")
+	})
+}
+
+// CheckAgentRestarts runs tests to check the agent can restart properly
+func CheckAgentRestarts(t *testing.T, client *ExtendedClient) {
+
+	t.Run("restart when stopped", func(tt *testing.T) {
+		_, err := client.SvcManager.Stop("datadog-agent")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Restart("datadog-agent")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Status("datadog-agent")
+		require.NoError(tt, err, "datadog-agent should restart when stopped")
+	})
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Don't you need to wait for the state transition to complete after triggering the status change ? Is the `SvcManager` call blocking ?

Reply:
It should not be required since `systemctl` is supposed to blocking during the state transition. If it appears to be flaky we can add some wait to be sure the state is correctly updated